### ISSUE TITLE
Transitive dependencies aren't pinned by install_with_constraints in Ch 3

### ISSUE THREAD

**Issue by josephhaaga** [cjolowicz/hypermodern-python#163]
After completing the *Managing dependencies with Nox sessions in Poetry* section in [Chapter 3](https://cjolowicz.github.io/posts/hypermodern-python-03-linting/), I encounter an error about dependencies not being pinned. 

It looks like the dependencies of my dev dependencies (which are now installed via `install_with_constraints`) are not being pinned correctly

```bash
ERROR: In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
    bandit from https://files.pythonhosted.org/packages/5a/50/ff2f2c8f1f0ca1569f678eeb608c0f973b835985410985594fbee96be820/bandit-1.6.2-py2.py3-none-any.whl#sha256=336620e220cf2d3115877685e264477ff9d9abaeb0afe3dc7264f55fa17a3952 (from flake8-bandit==2.1.2->-c requirements-constrained.txt (line 47))
    flake8-polyfill from https://files.pythonhosted.org/packages/86/b5/a43fed6fd0193585d17d6faa7b85317d4461f694aaed546098c69f856579/flake8_polyfill-1.0.2-py2.py3-none-any.whl#sha256=12be6a34ee3ab795b19ca73505e7b55826d5f6ad7230d31b18e106400169b9e9 (from flake8-bandit==2.1.2->-c requirements-constrained.txt (line 47))
    pycodestyle from https://files.pythonhosted.org/packages/10/5b/88879fb861ab79aef45c7e199cae3ef7af487b5603dcb363517a50602dd7/pycodestyle-2.6.0-py2.py3-none-any.whl#sha256=2295e7b2f6b5bd100585ebcb1f616591b652db8a741695b3d8f5d28bdc934367 (from flake8-bandit==2.1.2->-c requirements-constrained.txt (line 47))
    attrs>=19.2.0 from https://files.pythonhosted.org/packages/14/df/479736ae1ef59842f512548bacefad1abed705e400212acba43f9b0fa556/attrs-20.2.0-py2.py3-none-any.whl#sha256=fce7fc47dfc976152e82d53ff92fa0407700c21acd20886a13777a0d20e655dc (from flake8-bugbear==20.1.4->-c requirements-constrained.txt (line 51))
    importlib-metadata; python_version < "3.8" from https://files.pythonhosted.org/packages/6d/6d/f4bb28424bc677bce1210bc19f69a43efe823e294325606ead595211f93e/importlib_metadata-2.0.0-py2.py3-none-any.whl#sha256=cefa1a2f919b866c5beb7c9f7b0ebb4061f30a8a9bf16d609b000e2dfaceb9c3 (from flake8==3.8.4->-c requirements-constrained.txt (line 57))
    pyflakes<2.3.0,>=2.2.0 from https://files.pythonhosted.org/packages/69/5b/fd01b0c696f2f9a6d2c839883b642493b431f28fa32b29abc465ef675473/pyflakes-2.2.0-py2.py3-none-any.whl#sha256=0d94e0e05a19e57a99444b6ddcf9a6eb2e5c68d3ca1e98e90707af8152c90a92 (from flake8==3.8.4->-c requirements-constrained.txt (line 57))
    mccabe<0.7.0,>=0.6.0 from https://files.pythonhosted.org/packages/87/89/479dc97e18549e21354893e4ee4ef36db1d237534982482c3681ee6e7b57/mccabe-0.6.1-py2.py3-none-any.whl#sha256=ab8a6258860da4b6677da4bd2fe5dc2c659cff31b3ee4f7f5d64e79735b80d42 (from flake8==3.8.4->-c requirements-constrained.txt (line 57))
    toml>=0.10.1 from https://files.pythonhosted.org/packages/9f/e1/1b40b80f2e1663a6b9f497123c11d7d988c0919abbf3c3f2688e448c5363/toml-0.10.1-py2.py3-none-any.whl#sha256=bda89d5935c2eac546d648028b9901107a595863cb36bae0c73ac804a9b4ce88 (from black==20.8b1->-c requirements-constrained.txt (line 1))
    pathspec<1,>=0.6 from https://files.pythonhosted.org/packages/5d/d0/887c58853bd4b6ffc7aa9cdba4fc57d7b979b45888a6bd47e4568e1cf868/pathspec-0.8.0-py2.py3-none-any.whl#sha256=7d91249d21749788d07a2d0f94147accd8f845507400749ea19c1ec9054a12b0 (from black==20.8b1->-c requirements-constrained.txt (line 1))
    mypy-extensions>=0.4.3 from https://files.pythonhosted.org/packages/5c/eb/975c7c080f3223a5cdaff09612f3a5221e4ba534f7039db34c35d95fa6a5/mypy_extensions-0.4.3-py2.py3-none-any.whl#sha256=090fedd75945a69ae91ce1303b5824f428daf5a028d2f6ab8a299250a846f15d (from black==20.8b1->-c requirements-constrained.txt (line 1))
    appdirs from https://files.pythonhosted.org/packages/3b/00/2344469e2084fb287c2e0b57b72910309874c3245463acd6cf5e3db69324/appdirs-1.4.4-py2.py3-none-any.whl#sha256=a841dacd6b99318a741b166adb07e19ee71a274450e68237b4650ca1055ab128 (from black==20.8b1->-c requirements-constrained.txt (line 1))
    typing-extensions>=3.7.4 from https://files.pythonhosted.org/packages/60/7a/e881b5abb54db0e6e671ab088d079c57ce54e8a01a3ca443f561ccadb37e/typing_extensions-3.7.4.3-py3-none-any.whl#sha256=7cb407020f00f7bfc3cb3e7881628838e69d8f3fcab2f64742a5e76b2f841918 (from black==20.8b1->-c requirements-constrained.txt (line 1))
    regex>=2020.1.8 from https://files.pythonhosted.org/packages/93/8c/17f45cdfb39b13d4b5f909e4b4c2917abcbdef9c0036919a0399769148cf/regex-2020.9.27.tar.gz#sha256=a6f32aea4260dfe0e55dc9733ea162ea38f0ea86aa7d0f77b15beac5bf7b369d (from black==20.8b1->-c requirements-constrained.txt (line 1))
    typed-ast>=1.4.0 from https://files.pythonhosted.org/packages/0c/f0/c351d6aeb7f5ba00da2e5aac43ef09b71be95aed45bfca943a5016854269/typed_ast-1.4.1-cp37-cp37m-macosx_10_9_x86_64.whl#sha256=8ce678dbaf790dbdb3eba24056d5364fb45944f33553dd5869b7580cdbb83614 (from black==20.8b1->-c requirements-constrained.txt (line 1))
nox > Session lint-3.7 failed.
```

Some info

```
poetry --version
Poetry version 1.1.0
```

```
pyenv --version
pyenv 1.2.20
```

```
# pyproject.toml
[tool.poetry]
name = "cheap-chef"
version = "0.1.0"
description = "Recipe recommendations based on coupons at your local grocery store"
authors = ["Joseph Haaga <haaga.joe@gmail.com>"]
license = "MIT"
readme = "README.md"
homepage = "https://github.com/josephhaaga/cheap-chef"
repository = "https://github.com/josephhaaga/cheap-chef"
keywords = ["cooking", "food"]

[tool.poetry.dependencies]
python = "^3.7"
click = "^7.1.2"
requests = "^2.24.0"

[tool.poetry.dev-dependencies]
pytest = "^6.1.0"
coverage = {version = "^5.3", extras = ["toml"]}
pytest-cov = "^2.10.1"
pytest-mock = "^3.3.1"
flake8 = "^3.8.3"
black = "^20.8b1"
flake8-black = "^0.2.1"
flake8-import-order = "^0.18.1"
flake8-bugbear = "^20.1.4"
flake8-bandit = "^2.1.2"
safety = "^1.9.0"

[tool.poetry.scripts]
cheap-chef = "cheap_chef.console:main"

[tool.coverage.paths]
source = ["src", "*/site-packages"]

[tool.coverage.run]
branch = true
source = ["cheap_chef"]

[tool.coverage.report]
show_missing = true
fail_under = 100

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

**Comment by cjolowicz** [comment]
Thanks for reporting! This appears to be a bug in Poetry 1.1.0. I would advise you to downgrade to 1.0.x for now.

**Comment by cjolowicz** [comment]
Related: #162

**Comment by cjolowicz** [comment]
The upstream issue is here: https://github.com/python-poetry/poetry/issues/3074

I'm closing this because it's an upstream bug, but feel free to continue the discussion.

**Comment by cjolowicz** [comment]
The upstream issue has been fixed in https://github.com/python-poetry/poetry/pull/3024, released in [Poetry 1.1.1](https://github.com/python-poetry/poetry/releases/tag/1.1.1).
